vtex-apps/product-customizer
# Hotfixes

#### What is the purpose of this pull request?
Hotfixes several issues of products not rendering because of their attachments config.

#### What problem is this solving?
**NOTE:** The right thing to do here is to refactor the app entirely. I planned to do that already, with some changes to the UI and purchase flow that would make sense to more types of products other than customizable pizzas, but couldn't get the time to do so.

For now, this includes hotfixes for:

1. Rendering `vtex.product-details` when the attachments don't exist or are not valid for product-customizer (Yes, it's a hack, I already talked with @brunojdo with plans to port this app to be inside product-details)
2. The first variation shown (which should be Crusts following the attachments config) was sometimes showing another variation. This hardcodes to find only schema properties that are type string (i.e. you can't select multiple), and gets only the first one. (yes, this should allow for more than one attachment configured like this, see NOTE above).
3. Don't render attachments that are incorrectly configured and show console warning like "The attachment configuration for {{sku}} seems to be incorrect, please review the attachment."
4. Remove the hardcoded need for it to have one variation to render Checkboxes and one of NumericSteppers ("composition" and "optional" as they are called now)
5. Fixes all eslint errors
6. Remove unused code that was left behind.

#### How should this be manually tested?
Check if the following products are rendering and are able to be selected and show the variations.

- https://prodcustompr--delivery.myvtex.com/super-suprema/p
- https://prodcustompr--delivery.myvtex.com/4-quesos/p
- https://prodcustompr--delivery.myvtex.com/pizza-hawaiian/p
- https://prodcustompr--delivery.myvtex.com/pizza-pepperoni/p
- https://prodcustompr--delivery.myvtex.com/create-your-own/p

Check if other products are rendering product-details
- https://prodcustompr--delivery.myvtex.com/coca-cola/p
- https://prodcustompr--delivery.myvtex.com/shake/p
- and others...

#### Types of changes
